sonic_platform_base/sonic_xcvr/api/xcvr_api.py

```diff
@@ -0,0 +1,65 @@
+"""
+    xcvr_api.py
+
+    Abstract base class for platform-independent APIs used to interact with
+    xcvrs in SONiC
+"""
+
+class XcvrApi(object):
+    def __init__(self, xcvr_eeprom):
+        self.xcvr_eeprom = xcvr_eeprom
+
+    def get_model(self):
+        raise NotImplementedError
+
+    def get_serial(self):
+        raise NotImplementedError
+
+    def get_transceiver_info(self):
+        raise NotImplementedError
+
+    def get_transceiver_bulk_status(self):
+        raise NotImplementedError
+
+    def get_transceiver_threshold_info(self):
+        raise NotImplementedError
+
+    def get_rx_los(self):
+        raise NotImplementedError
+
+    def get_tx_fault(self):
+        raise NotImplementedError
+
+    def get_tx_disable(self):
+        raise NotImplementedError
+
+    def get_tx_disable_channel(self):
+        raise NotImplementedError
+
+    def get_temperature(self):
+        raise NotImplementedError
+
+    def get_voltage(self):
+        raise NotImplementedError
+
+    def get_tx_bias(self):
```

Comment:
I think we might want to consider explicitly defining whether these APIs are per host (ASIC) lane or per media lane. For some optics there are multiple host lanes mapped to a single media lane (i.e. DR4+ has a 2:1 ratio combining 2 50G inputs to 1 100G optical link). In these cases, my experience is the module only reports 4 tx_bias values, 4 rx_power, etc. Likewise TX disable would apply to the media lane. I'd suggest that these should all be defined as being per host lane, with the underlying implementation taking care of mapping to media lane. (so for DR4+, host lanes 1 and 2 would both be mapped to the first media lane, and so on)

Reply:
I think we can come back to this in a future PR, e.g. when we explicitly want to add support for DR4+. This looks to be outside the scope of this refactor.